tutorials/20_Using_Haystack_with_REST_API.ipynb

```diff
@@ -0,0 +1,415 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Tutorial: Using Haystack with REST API\n",
+    "\n",
+    "- **Level**: Advanced\n",
+    "- **Time to complete**: 30 minutes\n",
+    "- **Prerequisites**: Basic understanding of Docker, basic knowledge of Haystack pipelines \n",
+    "- **Nodes Used**: `ElasticsearchDocumentStore`, `EmbeddingRetriever`\n",
+    "- **Goal**: Learn how you can interact with Haystack through REST API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "\n",
+    "Haystack enables you to apply the latest NLP technology to your own data and create production-ready applications. Building an end-to-end NLP application requires the combination of multiple concepts:\n",
+    "* **DocumentStore** is the component in Haystack responsible for loading and storing text data in form of [Documents](https://docs.haystack.deepset.ai/docs/documents_answers_labels#document). In this tutorial, the DocumentStore will use Elasticsearch behind the scene.\n",
+    "* **Haystack** pipelines convert files into Documents, index them to DocumentStore, and run NLP tasks such as question answering and document search.\n",
+    "* **REST API**, as a concept, allows applications to interact with each other by handling their queries and returning responses. There is `rest_api` application within Haystack that exposes Haystack's functionalities through a RESTful API.\n",
+    "* **Docker** simplifies the environment setup needed to have Elasticsearch and Haystack API running.\n",
+    "\n",
+    "This tutorial introduces you to all the concepts needed to build an end-to-end document search application. After completing this tutorial, you will have learned how to create a pipeline YAML file, index files, and how to query your pipeline using REST API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preparing the Environment"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Install [Docker Compose](https://docs.docker.com/compose/), and launch Docker.\n",
+    "\n",
+    "If you installed Docker Desktop, you just need to start the application. Run `docker info` to see if Docker is up and running.\n",
+    "\n",
+    "```bash\n",
+    "docker info\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. Clone Haystack repository.\n",
```

Comment:
Hi @masci, I agree that they don't need to clone the whole repository only for the compose file, yet, I am not sure how to mitigate this issue. In my opinion, if we make them create the compose file from scratch in the tutorial, we need to explain what each line of code does. I didn't want to do that because:
1. Explaining the compose file would make the tutorial longer and more complicated for inexperienced users
2. The focus point of this tutorial is the pipeline YAML file and how to edit it, I don't want to shift toward Docker and put it in a spotlight

That's why I think for this tutorial it's more important to provide a compose file that works out of the box. Maybe I can give the Github link to the compose file and ask them to copy/paste it instead of making them clone the repo. wdyt?

Reply:
Good point, they can `curl https://raw.githubusercontent.com/deepset-ai/haystack/main/docker-compose.yml` 👍